src/hooks/courses/useCoursePayment.js

```
@@ -0,0 +1,67 @@
+      // User is the creator of the course
+      session.user.pubkey === course.pubkey ||
+      // Course is free
+      !isPaidCourse ||
+      // User has purchased this specific course
+      session.user.purchased?.some(purchase => purchase.courseId === course.d)
+    );
+  }, [session, course, isPaidCourse]);
+  
+  // Handler for successful payment
+  const handlePaymentSuccess = useCallback(async (response) => {
+    if (response && response?.preimage) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)